core/catalog.go

```diff
@@ -290,3 +292,119 @@ func (c *Catalog) Transfers(time0, time1 string) []CatalogEntry {
 	}
 	return out
 }
+
+// Insert new request
+func (c *Catalog) InsertRequest(request TransferRequest) error {
+	stm := getSQL("insert_request")
+	_, e := DB.Exec(stm, request.Id, request.File, request.Block, request.Dataset, request.SrcUrl, request.DstUrl, "pending", request.Priority)
+	if e != nil {
+		if !strings.Contains(e.Error(), "UNIQUE") {
+			check("Unable to insert into datasets table", e)
+		}
+	}
+	return e
+}
+
+// Update the status of request
+func (c *Catalog) UpdateRequest(id int64, status string) error {
+	stm := getSQL("update_request")
+	_, err := DB.Exec(stm, status, id)
+	return err
+}
+
+// Get the request details based on request id
+func (c *Catalog) RetriveRequest(request *TransferRequest) error {
+	stm := getSQL("request_by_id")
+	rows, err := DB.Query(stm, request.Id)
+	if err != nil {
+		request.Status = err.Error()
+		return err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		if err := rows.Scan(&request.File, &request.Block, &request.Dataset, &request.SrcUrl, &request.DstUrl, &request.Priority); err != nil {
+			request.Status = err.Error()
+			return err
+		}
+	}
+	return nil
+}
+
+// Get the status of request
+func (c *Catalog) GetStatus(id int64) (error, string) {
```

Comment:
For style correctness please use (value, error) in return, here I would suggest to swap return values, but then you'll need to modify code which calls GetStatus.